src/python/TaskWorker/MasterWorker.py

```diff
@@ -392,6 +392,34 @@ def _lockWork(self, limit, getstatus, setstatus):
 
         return True
 
+    def runCanary(self):
+        # Decide whether to use canary_name based on the canary_load probability.
+        use_canary = random.random() < self.config.TaskWorker.canary_load
+        workername = (
+            self.config.TaskWorker.canary_name 
+            if use_canary 
+            else self.config.TaskWorker.name
+        )
+
+        # Build the configreq dictionary
+        configreq = {
+            'subresource': 'process',
+            'workername': workername
+        }
+
+        try:
+            self.crabserver.post(api='workflowdb', data=urlencode(configreq))
+        except HTTPException as hte:
+            msg = "HTTP Error during runCanary: %s\n" % str(hte)
```

Comment:
better a more specific message, like "Error trying to change TW name". I think that when we log exception there's always a traceback. No need to put the module name here.

Reply:
Changes made, thanks!